config/config.go

```diff
@@ -419,7 +417,7 @@ func defaultProxyWithClientID(clientID string) *Proxy {
 	c.Producer.RetryMax = 6
 	c.Producer.ShutdownTimeout = 30 * time.Second
 
-	c.Consumer.AckTimeout = 15 * time.Second
+	c.Consumer.AckTimeout = 45 * time.Second
```

Comment:
Why are the Ack timeouts so aggressive, If scout moves to a model of only ack after processing an event successfully we will hit this timeout in the case of a network outage or temporary disruption. In the case of a temporary outage, this will result in the event being processed twice by scout. It is better to have a longer timeout to mitigate such instances. 300 seconds is relatively standard for network communication. In scouts case gocql has a network "connectivity" timeout of 30 seconds, but will retry each node in the cluster with a 3 seconds connection timeout. Which means by the time gocql has returned an error, scout will log the event as un-processed and will not ack the event. scout will then ask pixy for the same event again and try processing the event again, all within the 300 seconds ack timeout window.

Also, shouldn't `c.Consumer.SubscriptionTimeout` be larger than the `AckTimeout` ?

Your thoughts please,

Reply:
We can make it 300 seconds, no problem. But note that we already override this default in Radar setting it to 90 seconds.

There used to be a requirement to make `SubscriptionTimeout` larger than `AckTimeout` for otherwise `AckTimeout` was not honored on rebalancing, but there is no such requirement anymore. A relation between the timeouts is as follows: if consume requests for a group-topic stop coming for `SubscritionTimeout`, then rebalancing will happen as soon as the last pending request is acked, or `AckTimeout` elapses. Until rebalancing is triggered any incoming requests will reset `SubscritionTimeout` and consumption of the group-topic will be resumed.

Note that Scout consumes events in auto-ack mode so `AckTimeout` is not used.